File: my-react-app/src/api/stats.ts
import { apiRequest } from "./client";

export interface StatsOverview {
  draft_surveys: number;
  active_surveys: number;
  completed_surveys: number;
  total_responses: number;
  activity_last_7_days: Array<{
    date: string;
    responses_count: number;
  }>;
}

export async function getOverview(): Promise<StatsOverview> {
  return apiRequest<StatsOverview>("/stats/overview");
}

File: app/schemas/stats.py
from __future__ import annotations

from pydantic import BaseModel


class DailyActivityItem(BaseModel):
    date: str
    responses_count: int


class StatsOverviewResponse(BaseModel):
    draft_surveys: int
    active_surveys: int
    completed_surveys: int
    total_responses: int
    activity_last_7_days: list[DailyActivityItem]

